test/e2e/v3/e2e_suite.go

```diff
@@ -60,272 +68,217 @@ var _ = Describe("kubebuilder", func() {
-			Expect(utils.UncommentCode(
-				filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
-				"#- webhookcainjection_patch.yaml", "#")).To(Succeed())
-			Expect(utils.UncommentCode(filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
-				`#- name: CERTIFICATE_NAMESPACE # namespace of the certificate CR
-#  objref:
-#    kind: Certificate
-#    group: cert-manager.io
-#    version: v1alpha2
-#    name: serving-cert # this name should match the one in certificate.yaml
-#  fieldref:
-#    fieldpath: metadata.namespace
-#- name: CERTIFICATE_NAME
-#  objref:
-#    kind: Certificate
-#    group: cert-manager.io
-#    version: v1alpha2
-#    name: serving-cert # this name should match the one in certificate.yaml
-#- name: SERVICE_NAMESPACE # namespace of the service
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: webhook-service
-#  fieldref:
-#    fieldpath: metadata.namespace
-#- name: SERVICE_NAME
-#  objref:
-#    kind: Service
-#    version: v1
-#    name: webhook-service`, "#")).To(Succeed())
-
-			By("building image")
-			err = kbc.Make("docker-build", "IMG="+kbc.ImageName)
-			Expect(err).Should(Succeed())
-
-			By("loading docker image into kind cluster")
-			err = kbc.LoadImageToKindCluster()
-			Expect(err).Should(Succeed())
-
-			// NOTE: If you want to run the test against a GKE cluster, you will need to grant yourself permission.
-			// Otherwise, you may see "... is forbidden: attempt to grant extra privileges"
-			// $ kubectl create clusterrolebinding myname-cluster-admin-binding \
-			// --clusterrole=cluster-admin --user=myname@mycompany.com
-			// https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control
-			By("deploying controller manager")
-			err = kbc.Make("deploy", "IMG="+kbc.ImageName)
-			Expect(err).Should(Succeed())
-
-			By("validate the controller-manager pod running as expected")
-			verifyControllerUp := func() error {
-				// Get pod name
-				podOutput, err := kbc.Kubectl.Get(
-					true,
-					"pods", "-l", "control-plane=controller-manager",
-					"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
-						"{{ \"\\n\" }}{{ end }}{{ end }}")
-				Expect(err).NotTo(HaveOccurred())
-				podNames := utils.GetNonEmptyLines(podOutput)
-				if len(podNames) != 1 {
-					return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
-				}
-				controllerPodName = podNames[0]
-				Expect(controllerPodName).Should(ContainSubstring("controller-manager"))
-
-				// Validate pod status
-				status, err := kbc.Kubectl.Get(
-					true,
-					"pods", controllerPodName, "-o", "jsonpath={.status.phase}")
-				Expect(err).NotTo(HaveOccurred())
-				if status != "Running" {
-					return fmt.Errorf("controller pod in %s status", status)
+		Context("and plugin go.kubebuilder.io/v2", func() {
+			It("should generate a runnable project", func() {
+				GenerateV2(kbc)
+				Run(kbc)
+			})
+		})
```

Comment:
Are we creating a project with the v2 plugin here? Why we need to do that in the test/e2e/v3/e2e_suite.go ?

Reply:
v2 plugin, 3-alpha project version.

Reply:
I catcher your intention and motivation 👍